atc/config.go

```diff
@@ -22,6 +23,33 @@ type Config struct {
 	Jobs          JobConfigs      `json:"jobs,omitempty"`
 }
 
+func UnmarshalConfig(payload []byte, config interface{}) error {
+	// a 'skeleton' of Config, specifying only the toplevel fields
+	type skeletonConfig struct {
+		Groups        interface{} `json:"groups,omitempty"`
+		Resources     interface{} `json:"resources,omitempty"`
+		ResourceTypes interface{} `json:"resource_types,omitempty"`
+		Jobs          interface{} `json:"jobs,omitempty"`
+	}
+
+	var stripped skeletonConfig
+	err := yaml.UnmarshalStrict(payload, &stripped)
```

Comment:
(replying to your other comment in this context instead)

Ah, you're right. It looks like `UnmarshalStrict` actually enforces disallowing unknown fields:

https://github.com/kubernetes-sigs/yaml/blob/4cd0c284b15f1735b8cc247df097d262b8903f9f/yaml.go#L42

I only kept this as `UnmarshalStrict` because the original code did that, but now that it's unmarshalling into a struct with expected fields it'll actually blow up on unknown toplevel keys.

I think we can just replace this with a `yaml.Unmarshal`, then:

```suggestion
	err := yaml.Unmarshal(payload, &stripped)
```